### PR TITLE
[Keyboard] Add F.Me Macropad

### PR DESCRIPTION
**Pull request by thompson-ele**
## Description
Added the F.Me macropad to QMK including default and VIA keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by thompson-ele** [comment]
Hello! I've added the readme. Sorry I totally missed that when I was going through the PR checklist

File: keyboards/keebzdotnet/fme/config.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B5A
+#define PRODUCT_ID      0x8008
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    keebzdotnet
+#define PRODUCT         keyboard
+#define DESCRIPTION     FMe
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/config.h:28]:
```suggestion
```

File: keyboards/keebzdotnet/fme/config.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B5A
+#define PRODUCT_ID      0x8008
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    keebzdotnet
+#define PRODUCT         keyboard
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/config.h:27]:
```suggestion
#define PRODUCT         FMe
```

File: keyboards/keebzdotnet/fme/config.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 0
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/config.h:49]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/keebzdotnet/fme/config.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/config.h:63]:
```suggestion
```

File: keyboards/keebzdotnet/fme/fme.c [hunk 5]
```diff
@@ -0,0 +1 @@
+#include "fme.h"
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/fme.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/keebzdotnet/fme/rules.mk [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/rules.mk:11]:
```suggestion
```

File: keyboards/keebzdotnet/fme/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/rules.mk:27]:
```suggestion
BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes	# Mouse keys
EXTRAKEY_ENABLE = yes	# Audio control and System control
CONSOLE_ENABLE = no	# Console for debug
COMMAND_ENABLE = no    # Commands for debug and configuration
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keebzdotnet/fme/fme.c (hunk 5 above)
Comment (thompson-ele) [reply keyboards/keebzdotnet/fme/fme.c:17]: @drashna I've added the GPL2+ license to this file and reviewed / committed your other change requests 👍

File: keyboards/keebzdotnet/fme/readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+![F Me macropad](https://i.imgur.com/FKD43mB.jpg)
+
+A macropad designed after a meme board seen on Reddit.
+
+* Keyboard Maintainer: [keebzdotnet](https://github.com/keebzdotnet)
+* Hardware Supported: F-Me Macropad
+* Hardware Availability: https://www.keebz.net/shop/p/f-me
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/keebzdotnet/fme/readme.md:11]: Could you add instructions on how to jump to the bootloader, to flash new firmware?

Comment (thompson-ele) [reply keyboards/keebzdotnet/fme/readme.md:15]: @drashna I've added instructions for the bootloader :)

**Review by drashna — APPROVED** [review]

File: keyboards/keebzdotnet/fme/rules.mk [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/keebzdotnet/fme/rules.mk:20]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by thompson-ele** [comment]
@fauxpark I've updated the rules.mk file with your changes. thanks!

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**